### ISSUE TITLE
Forbid too long call chains

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#842]
# Rule request

## Thesis

In case you work with a really strange API that uses a lot of nested decorator functions (or with lots of partial application and curing) you can end up with something like this in your code base:

```python
some(a)(b)(c)(d)
```

Well, that's not pythonic. This is a complexity rule to limit this kind of situations.

What do we need?
1. Option to regulate the complexity threshold, let's name it `max_call_level`
2. Default value should be `2`, where `3` calls will result in a violation
3. A new complexity violation and a visitor

Related: https://github.com/wemake-services/wemake-python-styleguide/blob/8e759c1e80370234ac7bb30cf5f97fc953841688/wemake_python_styleguide/violations/complexity.py#L576
Related: #737 (where almost the same feature is implemented)

**Comment by lucaionescu** [comment]
Hi, can I be assigned to this issue?

**Comment by sobolevn** [comment]
@lucaionescu sure! Thanks a lot for your help!

**Comment by sobolevn** [comment]
Hi, @lucaionescu! How's it going? Do you need any help?

**Comment by lucaionescu** [comment]
Hey, thanks for checking in, it's all good for now! I was diving into the documentation and the contribution guide for the past days, just wanted to make sure I understand everything before I start to code.
